ReSharper.FSharp/src/FSharp.Fantomas.Host/src/FantomasCodeFormatter.cs

```diff
@@ -1,62 +1,121 @@
           CodeFormatter.FormatDocumentAsync(args.FileName, SourceOrigin.SourceOrigin.NewSourceString(args.Source),
             Convert(args.FormatConfig), Convert(args.ParsingOptions), myChecker), null, null)
         .Result.Replace("\r\n", args.NewLineText);
 
-    private static Range Convert(RdFcsRange range) =>
-      CodeFormatter.MakeRange(range.FileName, range.StartLine, range.StartCol, range.EndLine, range.EndCol);
-
     private static FSharpParsingOptions Convert(RdFcsParsingOptions options) =>
       new FSharpParsingOptions(new[] { options.LastSourceFile },
         ListModule.OfArray(options.ConditionalCompilationDefines), DefaultDiagnosticOptions,
-        false, options.LightSyntax, false, options.IsExe);
-
-    private static FormatConfig.FormatConfig Convert(RdFantomasFormatConfig config) =>
-      new FormatConfig.FormatConfig(config.IndentSize, config.MaxLineLength, config.SemicolonAtEndOfLine,
-        config.SpaceBeforeParameter, config.SpaceBeforeLowercaseInvocation, config.SpaceBeforeUppercaseInvocation,
-        config.SpaceBeforeClassConstructor, config.SpaceBeforeMember, config.SpaceBeforeColon, config.SpaceAfterComma,
-        config.SpaceBeforeSemicolon, config.SpaceAfterSemicolon, config.IndentOnTryWith, config.SpaceAroundDelimiter,
-        config.MaxIfThenElseShortWidth, config.MaxInfixOperatorExpression, config.MaxRecordWidth,
-        config.MaxRecordNumberOfItems, Convert(config.RecordMultilineFormatter), config.MaxArrayOrListWidth,
-        config.MaxArrayOrListNumberOfItems, Convert(config.ArrayOrListMultilineFormatter),
-        config.MaxValueBindingWidth, config.MaxFunctionBindingWidth, config.MaxDotGetExpressionWidth,
-        config.MultilineBlockBracketsOnSameColumn, config.NewlineBetweenTypeDefinitionAndMembers,
-        config.KeepIfThenInSameLine, config.MaxElmishWidth, config.SingleArgumentWebMode,
-        config.AlignFunctionSignatureToIndentation, config.AlternativeLongMemberDefinitions,
-        config.MultiLineLambdaClosingNewline, config.DisableElmishSyntax, DefaultFormatConfig.EndOfLine,
-        config.KeepIndentInBranch, config.BlankLinesAroundNestedMultilineExpressions,
-        config.BarBeforeDiscriminatedUnionDeclaration, config.StrictMode);
-
-    private static FormatConfig.MultilineFormatterType Convert(string setting) => setting switch
+        false, options.LightSyntax ?? FSharpOption<bool>.None, false, options.IsExe);
+
+    private static FormatConfig Convert(string[] riderFormatConfigValues)
+    {
+      var riderFormatConfigDict =
+        EditorConfigFields
+          .Zip(riderFormatConfigValues,
+            (field, valueData) =>
+              (field.Name, Value: valueData == ""
+                ? field.Value
+                : field.Value switch
+                {
+                  int _ => int.Parse(valueData),
+                  bool _ => bool.Parse(valueData),
+                  { } x when IsMultilineFormatterType(x) => Convert(valueData),
```

Comment:
Is this the only non bool/int type parameter?

Reply:
Btw, how is it going to work with similar fields added in future?

Reply:
> Is this the only non bool/int type parameter?

except for EndOfLineStyle, yes

> Btw, how is it going to work with similar fields added in future?

converting a string to an Enum requires specifying the type, so there are no easy ways

Reply:
Aren't the types known after loading a given Fantomas assembly?